### PR TITLE
[ch30655] amend indicator radio creation

### PR DESCRIPTION
**Pull request by domdinicola**
[ch30655]

### CODE REVIEW

**Comment by domdinicola** [comment]
Manual Script `IndicatorBlueprint.objects.filter(display_type=IndicatorBlueprint.RATIO).update(calculation_formula_across_locations='latest')`

**Review by emaciupe — APPROVED** [review]
looks good!

File: django_api/etools_prp/apps/unicef/tasks.py [hunk 1]
```diff
@@ -406,6 +406,9 @@ def process_programme_documents(fast=False, area=False):
                                         elif i['unit'] == 'number':
                                             i['display_type'] = 'number'
 
+                                        if 'display_type' in i and i['display_type'] == 'ratio':
+                                            i['calculation_formula_across_locations'] = 'latest'
```
Comment (acory) [inline django_api/etools_prp/apps/unicef/tasks.py:410]: Pls re-check this change, as the card mentions calculation formula across `reporting periods`

Comment (domdinicola) [reply django_api/etools_prp/apps/unicef/tasks.py]: thx @acory :)

**Comment by robertavram** [comment]
> calculation_formula_across_locations

calculation formula across reporting periods

**Comment by domdinicola** [comment]
I changed that @acory @robertavram  thx!

**Comment by robertavram** [comment]
This PR does not address the core of what is being asked. This card is actually related to [CH29826] 

https://app.shortcut.com/unicefetools/story/29826/prp-indicator-calculation-across-time-methods-should-include-latest


We need to add the option to select "Latest" as a calculation method across periods. Then we need to update the function that calculates this value to take into consideration the calculation method (right now it doesn't) `apps>indicator>models:L991`
The options need to be available on the model as well. (IndicatorBlueprint)

File: django_api/etools_prp/apps/indicator/models.py [hunk 2]
```diff
@@ -999,6 +1001,8 @@ def recalculate_reportable_total(sender, instance, **kwargs):
                     accepted_indicator_reports,
                     key=lambda item: item.total['v'])
                 reportable_total = max_total_ir.total
+            elif blueprint.calculation_formula_across_periods == IndicatorBlueprint.LATEST:
+                reportable_total = accepted_indicator_reports.last().total
```
Comment (robertavram) [inline django_api/etools_prp/apps/indicator/models.py:1005]: latest_accepted_indicator_report = accepted_indicator_reports.order_by('-time_period_start').first()

File: django_api/etools_prp/apps/unicef/tasks.py [hunk 3]
```diff
@@ -406,6 +406,9 @@ def process_programme_documents(fast=False, area=False):
                                         elif i['unit'] == 'number':
                                             i['display_type'] = 'number'
 
+                                        if 'display_type' in i and i['display_type'] == 'ratio':
```
Comment (robertavram) [inline django_api/etools_prp/apps/unicef/tasks.py:409]: percentage is taking the latest as well. so if `i['unit'] == "percentage"` you can set the calculation_formula_across_locations to 'latest'

File: django_api/etools_prp/apps/indicator/models.py [hunk 4]
```diff
@@ -999,6 +1001,8 @@ def recalculate_reportable_total(sender, instance, **kwargs):
...
             else:   # if its SUM or avg then add data up
                 for indicator_report in accepted_indicator_reports:
                     reportable_total['v'] += indicator_report.total['v']
```
Comment (robertavram) [inline django_api/etools_prp/apps/indicator/models.py:1008]: on L1018 it looks like both ratio and percentage are always calculated as latest across time periods. We should make sure that we can't set any other calculation methods for these indicators from the FE. If we will need to, then we'll have to come back and determine how that works.

**Review by robertavram — CHANGES_REQUESTED** [review]
changes

**Pull request merged**